crates/router/src/compatibility/stripe/customers/types.rs

```diff
@@ -1,34 +1,86 @@
 use std::{convert::From, default::Default};
 
-use api_models::payment_methods as api_types;
+use api_models::{customers, payment_methods as api_types};
 use common_utils::{
     crypto::Encryptable,
     date_time,
     pii::{self, Email},
 };
+use masking::{Secret, SerializableSecret};
 use serde::{Deserialize, Serialize};
 
-use crate::{logger, types::api};
+use crate::{
+    logger,
+    types::{api, api::enums as api_enums},
+};
+
+impl SerializableSecret for AddressDetails {}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct Shipping {
+    pub address: AddressDetails,
+    pub name: Option<Secret<String>>,
+    pub carrier: Option<String>,
+    pub phone: Option<Secret<String>>,
+    pub tracking_number: Option<Secret<String>>,
+}
+
+#[derive(Default, Serialize, PartialEq, Eq, Deserialize, Clone)]
+pub struct AddressDetails {
+    pub city: Option<String>,
+    pub country: Option<api_enums::CountryAlpha2>,
+    pub line1: Option<Secret<String>>,
+    pub line2: Option<Secret<String>>,
+    pub postal_code: Option<Secret<String>>,
+    pub state: Option<Secret<String>>,
+}
 
 #[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
 pub struct CreateCustomerRequest {
     pub email: Option<Email>,
     pub invoice_prefix: Option<String>,
-    pub name: Option<masking::Secret<String>>,
-    pub phone: Option<masking::Secret<String>>,
-    pub address: Option<masking::Secret<serde_json::Value>>,
+    pub name: Option<Secret<String>>,
+    pub phone: Option<Secret<String>>,
+    pub address: Option<Secret<AddressDetails>>,
     pub metadata: Option<pii::SecretSerdeValue>,
     pub description: Option<String>,
+    pub shipping: Option<Shipping>,
+    pub payment_method: Option<String>,              // not used
+    pub balance: Option<i64>,                        // not used
+    pub cash_balance: Option<pii::SecretSerdeValue>, // not used
+    pub coupon: Option<String>,                      // not used
+    pub invoice_settings: Option<pii::SecretSerdeValue>, // not used
+    pub next_invoice_sequence: Option<String>,       // not used
+    pub preferred_locales: Option<String>,           // not used
+    pub promotion_code: Option<String>,              // not used
+    pub source: Option<String>,                      // not used
+    pub tax: Option<pii::SecretSerdeValue>,          // not used
+    pub tax_exempt: Option<String>,                  // not used
+    pub tax_id_data: Option<String>,                 // not used
+    pub test_clock: Option<String>,                  // not used
 }
 
 #[derive(Clone, Default, Serialize, Deserialize, PartialEq, Eq)]
 pub struct CustomerUpdateRequest {
     pub description: Option<String>,
     pub email: Option<Email>,
-    pub phone: Option<masking::Secret<String, masking::WithType>>,
-    pub name: Option<masking::Secret<String>>,
-    pub address: Option<masking::Secret<serde_json::Value>>,
+    pub phone: Option<Secret<String, masking::WithType>>,
```

Comment:
if you are changing this, you don't have to mention `masking::WithType` it's auto-inferred.

Reply:
Okay